Dockerfile

```diff
@@ -46,46 +48,40 @@ RUN apt-get update -y -q && apt-get upgrade -y -q && apt-get upgrade -y -q && \
     curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
     unzip awscliv2.zip && \
     ./aws/install && \
-    rm -rf aws*
+    rm -rf aws* && \
+    cd /opt && \
+    curl "https://compiler-explorer.s3.amazonaws.com/opt/gcc-12.1.0.tar.xz" -o gcc12.tar.xz && \
+    tar Jxvf gcc12.tar.xz && \
+    rm gcc12.tar.xz
 
-# Install GCC 11 from the Ubuntu test repository
-RUN add-apt-repository ppa:ubuntu-toolchain-r/test -y && \
-    apt-get update && \
-    apt-get install -y -q gcc-11 g++-11 gnat-11 && \
-    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 60 --slave /usr/bin/g++ g++ /usr/bin/g++-11 && \
-    update-alternatives --config gcc
+## Need for host GCC version to be >= latest cross GCC being built.
+## This is at least needed for building cross-GNAT (Ada) as the GNAT runtime has no
+## requirement on a minimal supported version (e.g. need GCC 12 to build any GNAT runtime).
+## This is only true for cross compiler. Native compiler can use host's runtime
+## and bootstrap everything.
+ENV PATH="/opt/gcc-12.1.0/bin:${PATH}"
```

Comment:
I'm wondering if gcc-12 shouldn't be installed into /opt/compiler-explorer/gcc-12.1.0 in case it produces executables with references to .so's

Reply:
Good point, can't hurt, will do that!

Reply:
done and tested!